From e9e506ed436859048f6efc3b5962c6809f1a592a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Tue, 17 Sep 2013 15:13:18 -0500
Subject: [PATCH 153/169] Make test-login and test-sleep output debugging

Without a call to log_parse_environment(), things
like SYSTEMD_LOG_LEVEL do not work.
---
 src/login/test-login.c |  9 ++++++++-
 src/test/test-sleep.c  | 12 +++++++++++-
 2 files changed, 19 insertions(+), 2 deletions(-)

diff --git a/src/login/test-login.c b/src/login/test-login.c
index 945cb38..228ddb2 100644
--- a/src/login/test-login.c
+++ b/src/login/test-login.c
@@ -27,7 +27,7 @@
 #include "util.h"
 #include "strv.h"
 
-int main(int argc, char* argv[]) {
+static void test_login(void) {
         int r, k;
         uid_t u, u2;
         char *seat, *type, *class, *display;
@@ -215,6 +215,13 @@ int main(int argc, char* argv[]) {
         }
 
         sd_login_monitor_unref(m);
+}
+
+int main(int argc, char* argv[]) {
+        log_parse_environment();
+        log_open();
+
+        test_login();
 
         return 0;
 }
diff --git a/src/test/test-sleep.c b/src/test/test-sleep.c
index 545dfab..a1020ad 100644
--- a/src/test/test-sleep.c
+++ b/src/test/test-sleep.c
@@ -29,7 +29,7 @@
 #include "sleep-config.h"
 #include "strv.h"
 
-int main(int argc, char* argv[]) {
+static void test_sleep(void) {
         _cleanup_strv_free_ char
                 **standby = strv_new("standby", NULL),
                 **mem = strv_new("mem", NULL),
@@ -52,6 +52,16 @@ int main(int argc, char* argv[]) {
         log_info("Suspend configured and possible: %s", yes_no(can_sleep("suspend") > 0));
         log_info("Hibernation configured and possible: %s", yes_no(can_sleep("hibernate") > 0));
         log_info("Hybrid-sleep configured and possible: %s", yes_no(can_sleep("hybrid-sleep") > 0));
+}
+
+int main(int argc, char* argv[]) {
+        log_parse_environment();
+        log_open();
+
+        if (getuid() != 0)
+                log_warning("This program is unlikely to work for unpriviledged users");
+
+        test_sleep();
 
         return 0;
 }
-- 
1.8.4

